costmap_2d/src/footprint.cpp

```diff
@@ -137,12 +137,29 @@ void transformFootprint(double x, double y, double theta, const std::vector<geom
 
 void padFootprint(std::vector<geometry_msgs::Point>& footprint, double padding)
 {
-  // pad footprint in place
-  for (unsigned int i = 0; i < footprint.size(); i++)
+  geometry_msgs::Point footprint_center;
+  double sum_x = 0.0, sum_y = 0.0;
+
+  for (const auto& point : footprint)
+  {
+    sum_x += point.x;
+    sum_y += point.y;
+  }
+  footprint_center.x = sum_x / footprint.size();
+  footprint_center.y = sum_y / footprint.size();
```

Comment:
S: I think it makes sense to create a method `centroid` that returns `geometry_msgs::Point`